### PR TITLE
fix: Surface error messages properly from the scanner

### LINKED ISSUE #1315: Abstract error message thrown when crawler encounters browser errors
## Describe the bug

When the crawler encounters a browser error, it will often throw the following error:

```
[error][Exception]ErrorWithCause: An error occurred while scanning website page
....
caused by: TypeError: Cannot read properties of undefined (reading 'baseUrl')
    at Scanner.getCombinedReportParameters (/home/vsts/work/_tasks/accessibility-insights_4811a442-2fd3-5aa8-ba1a-14cb7e24c113/3.0.0/index.js:191027:60)
    at Scanner.<anonymous> (/home/vsts/work/_tasks/accessibility-insights_4811a442-2fd3-5aa8-ba1a-14cb7e24c113/3.0.0/index.js:191005:55)
    at Generator.next (<anonymous>)
    at fulfilled (/home/vsts/work/_tasks/accessibility-insights_4811a442-2fd3-5aa8-ba1a-14cb7e24c113/3.0.0/index.js:190928:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

I believe this is happening because the `combinedScanResult` in `scanner.ts` returns an `errors` array which does not get evaluated and elevated to the logger. As a result, the `getCombinedReportParameters` attempts to create the report metadata, but because it doesn't exist, it fails when it attempts to reach the `baseUrl`. The error message thrown is abstracted from the actual problem - the crawler isn't getting a page response.

**To Reproduce**

I created an [example build](https://dev.azure.com/accessibility-insights-private/Accessibility%20Insights%20(private)/_build/results?buildId=37895&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=7384d774-f7ca-599c-ee57-ab2c05be9247&s=96ac2280-8cb4-5df5-99de-dd2da759617d)

## CodePen repro example

n/a

## Expected behavior

The scanner evaluates the `errors` array and called `this.logger.error()` for each error. This will return an improved error message from the scanner: `Unable to get a page response from the browser` 

Prior art from the service:
- [handles errors array](https://github.com/microsoft/accessibility-insights-service/blob/171bc5eff976890d54cc6cb53497ea1acbb0a389/packages/cli/src/runner/crawler-command-runner.ts#L43-L52)
- [test case with browser error](https://github.com/microsoft/accessibility-insights-service/blob/791716fc36fa2d8c68a2946cfb4ba05a38e9a128/packages/scanner-global-library/src/page-navigator.spec.ts#L298-L323)

## Context (please complete the following information)

- OS Name & Version: 
- AI-Action Version & Environment: ADO@v3
- Browser Version: 
- Target Page: 

## Are you willing to submit a PR?

Yes

## Additional context

@jlperkins and I debugged two different pipelines with this error over the past week.

### PR DESCRIPTION
**Pull request by brocktaylor7**
#### Details

This fixes an issue causing misleading error messages to appear when configuration issues or other errors cause the accessibility-insights-scan package to return an error.

When an error is surfaced from the scan package it returns it in `CombinedScanResults.errors` with empty `CombinedReportParameters`. We were previously not handling this scenario, so when the next step of the code was attempting to get the `baseUrl` from the `CombinedReportParameters`, it would fail and throw an ambiguous error message about the `baseUrl` that had nothing to do with the actual error.

This adds a check for the `CombinedScanResults.errors` array to ensure it's empty, and if it isn't it surfaces any errors returned in the array and fails the run.

This issue was occuring in the `shared` package that is consumed by both the ADO Extension and the GH Action, so this will remedy the issue in both products.

Previous error message example when a navigation error had occurred:
```text
##[error][Exception]ErrorWithCause: An error occurred while scanning website page https://test-site
    at Logger.trackExceptionAny (/home/brock/projects/ai-action/packages/ado-extension/dist/pkg/index.js:189142:29)
    at Scanner.<anonymous> (/home/brock/projects/ai-action/packages/ado-extension/dist/pkg/index.js:190329:29)
    at Generator.next (<anonymous>)
    at fulfilled (/home/brock/projects/ai-action/packages/ado-extension/dist/pkg/index.js:190244:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
caused by: TypeError: Cannot read properties of undefined (reading 'baseUrl')
    at Scanner.getCombinedReportParameters (/home/brock/projects/ai-action/packages/ado-extension/dist/pkg/index.js:190343:60)
    at Scanner.<anonymous> (/home/brock/projects/ai-action/packages/ado-extension/dist/pkg/index.js:190321:55)
    at Generator.next (<anonymous>)
    at fulfilled (/home/brock/projects/ai-action/packages/ado-extension/dist/pkg/index.js:190244:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

New error message example when a navigation error has occurred:
```text
Scan failed with 1 error(s)
##[error][Exception]ErrorWithCause: An error occurred while scanning website page: https://test-site
    at Logger.trackExceptionAny (/home/brock/projects/ai-action/packages/ado-extension/dist/pkg/index.js:189142:29)
    at Scanner.logAndTrackScanningException (/home/brock/projects/ai-action/packages/ado-extension/dist/pkg/index.js:190373:21)
    at /home/brock/projects/ai-action/packages/ado-extension/dist/pkg/index.js:190323:30
    at Array.forEach (<anonymous>)
    at Scanner.<anonymous> (/home/brock/projects/ai-action/packages/ado-extension/dist/pkg/index.js:190322:47)
    at Generator.next (<anonymous>)
    at fulfilled (/home/brock/projects/ai-action/packages/ado-extension/dist/pkg/index.js:190244:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
caused by: Error: '{"errorType":"NavigationError","message":"Unable to get a page response from the browser.","stack":"Error\\n    at PageNavigationHooks.postNavigation (/home/brock/projects/ai-action/packages/ado-extension/dist/pkg/node_modules/accessibility-insights-scan/dist/index.js:6551:24)\\n    at PageProcessorBase.postNavigation (/home/brock/projects/ai-action/packages/ado-extension/dist/pkg/node_modules/accessibility-insights-scan/dist/index.js:4306:48)\\n    at PuppeteerCrawler._executeHooks (/home/brock/projects/ai-action/packages/ado-extension/dist/pkg/node_modules/apify/build/crawlers/basic_crawler.js:592:23)\\n    at PuppeteerCrawler._handleNavigation (/home/brock/projects/ai-action/packages/ado-extension/dist/pkg/node_modules/apify/build/crawlers/browser_crawler.js:405:20)\\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\\n    at async PuppeteerCrawler._handleRequestFunction (/home/brock/projects/ai-action/packages/ado-extension/dist/pkg/node_modules/apify/build/crawlers/browser_crawler.js:343:13)\\n    at async wrap (/home/brock/projects/ai-action/packages/ado-extension/dist/pkg/node_modules/@apify/timeout/index.js:73:27)"}'
    at Logger.trackExceptionAny (/home/brock/projects/ai-action/packages/ado-extension/dist/pkg/index.js:189141:94)
    at Scanner.logAndTrackScanningException (/home/brock/projects/ai-action/packages/ado-extension/dist/pkg/index.js:190373:21)
    at /home/brock/projects/ai-action/packages/ado-extension/dist/pkg/index.js:190323:30
    at Array.forEach (<anonymous>)
    at Scanner.<anonymous> (/home/brock/projects/ai-action/packages/ado-extension/dist/pkg/index.js:190322:47)
    at Generator.next (<anonymous>)
    at fulfilled (/home/brock/projects/ai-action/packages/ado-extension/dist/pkg/index.js:190244:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

##### Motivation

Addresses issue #1315

#### Pull request checklist

- [x] Addresses an existing issue: Fixes #1315
- [x] Added relevant unit test for your changes. (`yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`yarn precheckin`)
- [x] Described how this PR impacts both the ADO extension and the GitHub action

### CODE REVIEW

File: packages/shared/src/scanner/scanner.ts [hunk 1]
```diff
@@ -94,6 +94,15 @@ export class Scanner {
 
             const scanStarted = new Date();
             const combinedScanResult = await this.crawler.crawl(crawlerParameters, this.baselineOptionsBuilder.build(scanArguments));
+
+            if (!isEmpty(combinedScanResult.errors)) {
+                this.logger.logInfo(`Scan failed with ${combinedScanResult.errors.length} error(s)`);
```
Comment (katydecorah) [inline packages/shared/src/scanner/scanner.ts:99]: What do you think about using `logError` here instead? With `logError` this message will elevate to the email notification and build summary.

```suggestion
                this.logger.logError(`Scan failed with ${combinedScanResult.errors.length} error(s)`);
```

**Review by madalynrose — APPROVED** [review]
This looks awesome!

**Review by katydecorah — APPROVED** [review]
Thanks, Brock! I left one small suggestion for you to consider.

File: packages/shared/src/scanner/scanner.ts (hunk 1 above)
Comment (brocktaylor7) [reply packages/shared/src/scanner/scanner.ts]: I like this better than the `logInfo` I was using. Good suggestion! I've updated to use `logError` instead. Thanks, Katy!

**Pull request merged**